logstash_async/transport.py

```diff
@@ -176,3 +227,144 @@ def send(self, events, use_logging=False):
+    logstash plugin `inputs_http`.
+
+    For more details visit:
+    https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html
+
+    :param host: The hostname of the logstash HTTP server.
+    :type host: str
+    :param port: The TCP port of the logstash HTTP server.
+    :type port: int
+    :param timeout: The connection timeout. (Default: None)
+    :type timeout: float
+    :param ssl_enable: Activates TLS. (Default: True)
+    :type ssl_enable: bool
+    :param ssl_verify: Activates the TLS certificate verification. If the flag
+    is True the class tries to verify the TLS certificate with certifi. If you
+    pass a string with a file location to CA certificate the class tries to
+    validate it against it. (Default: True)
+    :type ssl_verify: bool or str
+    :param username: Username for basic authorization. (Default: "")
+    :type username: str
+    :param password: Password for basic authorization. (Default: "")
+    :type password: str
+    :param max_content_length: The max content of an HTTP request in bytes.
+    (Default: 100MB)
+    :type max_content_length: int
+    """
+
+    def __init__(
+            self,
+            host,
+            port,
+            timeout=TimeoutNotSet,
+            ssl_enable=True,
+            ssl_verify=True,
+            **kwargs
+    ):
+        super().__init__(host, port, timeout, ssl_enable, ssl_verify)
+        self._username = kwargs.get('username', None)
+        self._password = kwargs.get('password', None)
+        self._max_content_length = kwargs.get('max_content_length', 104857600)
+        self.__session = None
+
+    @property
+    def url(self):
+        """The URL of the logstash pipeline based on the hostname, the port and
+        the TLS usage.
+
+        :return: The URL of the logstash HTTP pipeline.
+        :rtype: str
+        """
+        protocol = 'http'
+        if self._ssl_enable:
+            protocol = 'https'
+        return '{}://{}:{}'.format(protocol, self._host, self._port)
+
+    def __batches__(self, events):
+        """Generate dynamic sized batches based on the max content length.
+
+        :param events: A list of events.
+        :type events: list
+        :return: A list of event batches.
+        :rtype: collections.Iterable[list]
+        """
+        current_batch = []
+        event_iter = iter(events)
+        while True:
+            try:
+                current_event = next(event_iter)
+            except StopIteration:
+                current_event = None
+                if not current_batch:
+                    return
+                yield current_batch
+            if current_event is None:
+                return
+            if len(current_event) > self._max_content_length:
+                msg = 'The event size <%s> is greater than the max content length <%s>.'
+                msg += 'Skipping event.'
+                logger.warning(msg, len(current_event), self._max_content_length)
```

Comment:
Logging in transports can be tricky: if `logging.shutdown()` has been called, the logging framework basically doesn't work anymore and we have to just finish transmitting.
This is why the handler passes the `use_logging` argument to tell the transport whether it might use the logging framework or not.
See https://github.com/eht16/pylogbeat/blob/master/pylogbeat.py#L72 for some inspiration.

Reply:
Does the people use the AsynchronousLogstashHandler for the root logger or using the handler for the logstash_async itself? But you are right, it can be tricky. But I would not like to skip the event silently. Should we let it fail and just do not care about the corner case? The bad thing is that all future events will fail, too. Right?

Reply:
The idea is to use `AsynchronousLogstashHandler` in applications as a handler for the existing logging configuration and I guess usually for the root logger. So this handler primarily handles the application's log messages.

I agree to not silently drop events on errors. There is nothing wrong with this warning message but I wanted to suggest to issue it only if `use_logging` is set and to print to `stderr` otherwise.

It might be good to have `_safe_log` (https://github.com/eht16/python-logstash-async/blob/f09055373613c988f6ab4a9e485bb8b12598cafd/logstash_async/worker.py#L267) also available in the transports for this. But I can work on this after this PR, we don't need to refactor the world all in here :D.